src/DotNetWorker/Hosting/WorkerHostBuilderExtensions.cs

```diff
@@ -169,12 +169,19 @@ public static IHostBuilder ConfigureFunctionsWorkerDefaults(this IHostBuilder bu
                     // Call the provided configuration prior to adding default middleware
                     configure(context, appBuilder);
 
-                    // Add default middleware
-                    appBuilder.UseDefaultWorkerMiddleware();
-                })
-               .UseDefaultServiceProvider((context, options) =>
-                {
-                    options.ValidateScopes = context.HostingEnvironment.IsDevelopment();
+                    static bool ShouldSkipDefaultWorkerMiddleware(IDictionary<object, object> props)
+                    {
+                        return props is not null &&
+                            props.TryGetValue(FunctionsApplicationBuilder.SkipDefaultWorkerMiddlewareKey, out var skipObj) &&
+                            skipObj is bool skip &&
+                            skip;
```

Comment:
nit: should be able to update the pattern here to:
``` csharp
skipObj is true
```

Replacing lines 176 and 177.